src/API.ts

```diff
@@ -98,16 +114,51 @@ export function registerRoutes(server: FastifyInstance<Server, IncomingMessage,
         return
       }
       NodeList.toggleFirstNode()
-      const ip = signedFirstNodeInfo.nodeInfo.externalIp
-      const port = signedFirstNodeInfo.nodeInfo.externalPort
-      const publicKey = signedFirstNodeInfo.nodeInfo.publicKey
+
+      const signedFirstNodeInfo = requestBody.joinRequest.nodeInfo
+      const appJoinData = requestBody.joinRequest.appJoinData
+
+      const ip = signedFirstNodeInfo.externalIp
+      const port = signedFirstNodeInfo.externalPort
+      const publicKey = signedFirstNodeInfo.publicKey
 
       const firstNode: NodeList.ConsensusNodeInfo = {
         ip,
         port,
         publicKey,
       }
 
+      // eslint-disable-next-line no-constant-condition
+      if (config.VERBOSE) {
+        Logger.mainLogger.debug('POST /nodelist firstNode:', firstNode)
+        Logger.mainLogger.debug('POST /nodelist appJoinData:', appJoinData)
+      }
+      
+
+      const networkAccount: AccountDB.AccountsCopy | string = getGlobalNetworkAccount(false)
+      Logger.mainLogger.debug('networkAccount', networkAccount)
+
+      if (networkAccount) {
+        if (
+          typeof networkAccount != 'string' &&
+          !NodeList.isValidVersion(
+            networkAccount?.data?.current?.minVersion,
```

Comment:
How will the archiver knows the data type and format of the global/network account from the Shardeum? For instance, if archiver is used other app layers (other than Shardeum) that use different format for network account, this will fail.